app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/UpdateRecordRequest.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.request
+
+import play.api.libs.functional.syntax.{toApplicativeOps, toFunctionalBuilderOps}
+import play.api.libs.json.Format.GenericFormat
+import play.api.libs.json.Reads.verifying
+import play.api.libs.json.{JsPath, OWrites, Reads}
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.eis.Assessment
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ValidationSupport
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ValidationSupport.Reads.lengthBetween
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ValidationSupport.isValidCountryCode
+
+import scala.Function.unlift
+
+case class UpdateRecordRequest(
+  eori: String,
+  recordId: String,
+  actorId: String,
+  traderRef: Option[String],
+  comcode: Option[String],
+  goodsDescription: Option[String],
+  countryOfOrigin: Option[String],
+  category: Option[Int],
+  assessments: Option[Seq[Assessment]],
+  supplementaryUnit: Option[Int],
+  measurementUnit: Option[String],
+  comcodeEffectiveFromDate: Option[String],
```

Comment:
think comcodeEffectiveFromDate and comcodeEffectiveToDate they are not Instant type instead of String but not sure if we are going to address that as part of TGP-996 because it needs to updated everywhere in router.

Reply:
So I tested this using an Instant, and it parses it ok for the request but fails to parse it when EIS returns. I think we should leave it as is since it is passed and returned as a date string anyway

Reply:
I tested this again, and it's actually okay. The problem is the data that the stub sends back for the `createdDateTime` and `updatedDateTime`
```
    "createdDateTime": "2024-11-18T23->20->19Z",
    "updatedDateTime": "2024-11-18T23->20->19Z",
```
The data is invalid so we can have it as an `Instant`